core/src/main/java/org/apache/iceberg/rest/responses/LoadTableResponse.java

```diff
@@ -80,7 +81,24 @@ public TableMetadata tableMetadata() {
   }
 
   public Map<String, String> config() {
-    return config != null ? config : ImmutableMap.of();
+    Map<String, String> conf = config != null ? config : ImmutableMap.of();
+    List<Credential> storageCredentials = credentials();
+    if (!storageCredentials.isEmpty()) {
+      List<Credential> creds = Lists.newLinkedList(storageCredentials);
+      for (Credential credential : storageCredentials) {
+        // only keep the credentials with the longest prefix
+        creds.removeIf(
+            cred ->
+                credential.prefix().length() > cred.prefix().length()
+                    && credential.prefix().startsWith(cred.prefix()));
+      }
```

Comment:
Sorry not completely following, don't we need to keep all of these and dynamically choose the credential with the longest common prefix for a given file path? e.g. if I have `s3://foo/bar -> s3cred1` and `s3://foo -> s3cred2` and I have two paths `s3://foo/bar/file1.parquet` and `s3://foo/file2.paruqet`, the first one should use s3cred1 and the second one s3cred2

Reply:
I think it'd require more plumbing through FileIO. Or not sure if we're trying to solve something different here

Reply:
> I think it'd require more plumbing through FileIO. Or not sure if we're trying to solve something different here

In the long run we'd definitely need to do some plumbing so that e.g. S3FileIO is properly configured for different credential prefixes. I talked with @danielcweeks last week and the current goal here was to basically use the credentials that come back in the `storage-credentials` field in `LoadTableResponse` and merge those into the normal `config`, since in the spec we say that whatever in `storage-credentials` is should be used before looking into whatever is in the `config`

Reply:
You're right @amogh-jahagirdar, but the first step is to simply plumb through the storage credentials and then we can add support for multiple prefixes.

Reply:
Got it, the scope of the PR is just to handle passing through the credentials.